docs/content/TextInput.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: TextInput
+status: Alpha
+---
+
+import {PropsTable} from '../src/props-table'
+
+TextInput is a form component to add default styling to the native text input.
+
+**Note:** Don't forget to set `aria-label` to make the TextInput accessible to screen reader users.
+
+## Default example
+
+```jsx live
+<>
+  <TextInput aria-label="Zipcode" name="zipcode" placeholder="Zipcode" autoComplete="postal-code" />
+
+  <TextInput sx={{ml: 4}} aria-label="City" name="city" placeholder="City" contrast />
+
+  <TextInput
+    sx={{ml: 4}}
+    icon={SearchIcon}
+    aria-label="Zipcode"
+    name="zipcode"
+    placeholder="Find user"
+    autoComplete="postal-code"
+  />
+</>
+```
+
+## Component props
+
+Native `<input>` attributes are forwarded to the underlying React `input` component and are not listed below.
+
+<PropsTable>
+  <PropsTable.Row name="aria-label" type="string" description="Required. Allows input to be accessible." />
+  <PropsTable.Row name="block" type="boolean" description="Adds `display: block` to element" />
+  <PropsTable.Row name="contrast" type="boolean" description="Changes background color to a higher contrast color" />
+  <PropsTable.Row
+    name="variant"
+    type="string"
+    description="Can be either `small` or `large`. Creates a smaller or larger input than the default."
+  />
+  <PropsTable.Row name="width" type="string | number" description="Set the width of the input" />
+  <PropsTable.Row name="maxWidth" description="Set the maximum width of the input">
+    <PropsTable.Row.Type>
+      string | number | <Link href="https://styled-system.com/guides/array-props">Array</Link>
+    </PropsTable.Row.Type>
+  </PropsTable.Row>
```

Comment:
It feels a bit odd to me to have two ways of defining the type. I'd vote for a single `type` prop that accepts a string or a JSX element:

```
<PropsTable.Row ... type={<>string | number | <Link href="https://styled-system.com/guides/array-props">Array</Link></>} ...
```

Reply:
I think we can use the same batter for description too

Reply:
Yeah, I was experimenting a little bit here with the `createSlots` API, but something more straightforward should be fine.